integration/fuse/src/main/java/alluxio/fuse/AlluxioFuseFileSystemOpts.java

```diff
@@ -0,0 +1,230 @@
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.fuse;
+
+import alluxio.conf.AlluxioConfiguration;
+import alluxio.conf.PropertyKey;
+
+import com.google.common.base.Preconditions;
+import com.google.common.collect.ImmutableList;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+/**
+ * This class holds all fuse-related options used by libfuse and AlluxioFuse.
+ */
+public final class AlluxioFuseFileSystemOpts {
+  private final String mAlluxioPath;
+  private final String mFsName;
+  private final int mFuseMaxPathCached;
+  private final int mFuseUmountTimeout;
+  private final boolean mIsDebug;
+  private final List<String> mLibfuseOptions;
+  private final String mMountPoint;
+  private final boolean mSpecialCommandEnabled;
+  private final long mStatCacheTimeout;
+  private final boolean mUserGroupTranslationEnabled;
+
+  /**
+   * Constructs an {@link AlluxioFuseFileSystemOpts} with only Alluxio cluster configuration.
+   *
+   * @param conf     Alluxio cluster configuration
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(AlluxioConfiguration conf) {
+    List<String> libfuseOptions = conf.isSet(PropertyKey.FUSE_MOUNT_OPTIONS)
+        ? conf.getList(PropertyKey.FUSE_MOUNT_OPTIONS)
+        : ImmutableList.of();
+    return new AlluxioFuseFileSystemOpts(
+        conf.getString(PropertyKey.FUSE_MOUNT_ALLUXIO_PATH),
+        conf.getString(PropertyKey.FUSE_FS_NAME),
+        conf.getInt(PropertyKey.FUSE_CACHED_PATHS_MAX),
+        (int) conf.getMs(PropertyKey.FUSE_UMOUNT_TIMEOUT),
+        conf.getBoolean(PropertyKey.FUSE_DEBUG_ENABLED),
+        libfuseOptions,
+        conf.getString(PropertyKey.FUSE_MOUNT_POINT),
+        conf.getBoolean(PropertyKey.FUSE_SPECIAL_COMMAND_ENABLED),
+        conf.getMs(PropertyKey.FUSE_STAT_CACHE_REFRESH_INTERVAL),
+        conf.getBoolean(PropertyKey.FUSE_USER_GROUP_TRANSLATION_ENABLED)
+    );
+  }
+
+  /**
+   * Constructs an {@link AlluxioFuseFileSystemOpts} with
+   * Alluxio cluster configuration and command line.
+   * Command line input has higher precedence if a property is set both in config and command.
+   *
+   * @param conf     Alluxio cluster configuration
+   * @param fuseCliOpts     Alluxio fuse command line input
+   * @return AlluxioFuseFileSystemOpts
+   */
+  public static AlluxioFuseFileSystemOpts create(
+      AlluxioConfiguration conf, AlluxioFuseCliOpts fuseCliOpts) {
+    Optional<String> alluxioPathFromCli = fuseCliOpts.getMountAlluxioPath();
```

Comment:
alluxioPathFromCli is only used once so can be inline, no need to create a separate variable

Reply:
Surprisingly with the inline the method `orElseGet` on the next line makes a lot more sense.